### DISCUSSION TITLE
Fastapi Tutorials

### DISCUSSION THREAD

**Discussion by cosmos369** [fastapi/fastapi#2538] [Questions]
Hi,

I am new to programming(6 months approx). I have learned python as my go-to language. For development, I use flask and Django as my frameworks. I love Django as it is very structured and for a beginner like me, I am able to deploy a project completely using django.

Recently, one of my friend who is a very senior backend developer advised me to use fastapi which is very new in the market.

But I am not finding a structured tutorial on it as Django and flask has. If anyone has any links, pls send it to me as I love fastapi now.

**Comment by hard-coders** [comment]
🚀 Welcome aboard!
You can find lots of articles from [External Links and Articles](https://fastapi.tiangolo.com/external-links) 😎

**Reply by cosmos369** [reply]
is there a complete project tutorial by using fastapi...This will definitely increase the adoption of  fastapi as many beginners want that...moreover, in udemy or coursera, there is no single course on fastapi

**Reply by hard-coders** [reply]
Hmm...Well, I recommend [Test-Driven Development with FastAPI and Docker](https://testdriven.io/courses/tdd-fastapi/) course. It is provided by testdriven.io, which is [silver spornsor](https://fastapi.tiangolo.com/fastapi-people/#silver-sponsors) of FastAPI. But you need to purchase the course. It's beneficial, and I also purchased it already.

**Reply by tomsem** [reply]
https://training.talkpython.fm/courses/details/getting-started-with-fastapi and 'coming soon' **Full Web Apps with FastAPI**.

**Reply by koaning** [reply]
I'm obviously a bit biased, but there's also a mini-course for free on [calmcode](https://calmcode.io/fastapi/hello-world.html).

**Comment by morsley** [comment]
Besides the site's docu-tutorials, there isn't really a lot of external content and blogs on the framework yet.  I definitely feel required to dig into the source at times to understand things (particularly with auth topics).

**Comment by YuriiMotov** [comment]
I think at this moment FastAPI has quite comprehensive docs. And the amount of external resources is impressive!
Off course, there are still some topics that are not covered. Feel free to ask here in discussions!
